src/nvme/mi.h

```diff
@@ -955,6 +955,44 @@ int nvme_mi_admin_xfer(nvme_mi_ctrl_t ctrl,
 		       off_t resp_data_offset,
 		       size_t *resp_data_size);
 
+/**
+ * nvme_mi_admin_admin_passthru() - Submit an nvme admin passthrough command
+ * @ctrl: Controller to send command to
+ * @opcode:	The nvme admin command to send
+ * @flags:	NVMe command flags (not used)
+ * @rsvd:	Reserved for future use
+ * @nsid:	Namespace identifier
+ * @cdw2:	Command dword 2
+ * @cdw3:	Command dword 3
+ * @cdw10:	Command dword 10
+ * @cdw11:	Command dword 11
+ * @cdw12:	Command dword 12
+ * @cdw13:	Command dword 13
+ * @cdw14:	Command dword 14
+ * @cdw15:	Command dword 15
+ * @data_len:	Length of the data transferred in this command in bytes
+ * @data:	Pointer to user address of the data buffer
```

Comment:
You have a single buffer + length for both command and response data. However, it's possible for a command to be bi-directional.

Should we include a separate response buffer + length here? Then we don't need to guess the direction of the API from the command opcode.

(However, this will diverge from the existing direct ioctl API; I'm happy to discuss the better solution here. @igaw , how do you intend to handle bidirectional commands through the existing API?)

Reply:
Hmm, I was not involved when those APIs where designed. Maybe there was specific reason why there is only one buffer. Need to figure this out first.

But at this stage I think we should not start introducing different API styles for the pass through commands. As I understood the mi version of the functions are just using a different transport for the same commands. 

So this sounds like 2.x material.

Reply:
The nvme-cli subcommand `admin-passthru` doesn't support bidirectional data transfer for now(both in-band or OOB). So the discussion about bidirectional support is out of scope of this pull request.

Reply:
Yep, agreed. Let's keep this as-is.